### PR TITLE
🔧 fix ksm minting

### LINKED ISSUE #6916: Create token does not work on RMRK 1.0
RMRK1 test: 
https://kusama.subscan.io/tx/0x0680cf8826e0bbc283d052d6eaa06bf921c957b3c24cd62359d40c5808ea0aa4 ❌ 

_Originally posted by @prury in https://github.com/kodadot/nft-gallery/pull/6896#issuecomment-1689973609_

### LINKED ISSUE #7044: Artist unable to mint properly on RMRK2 (NFT not loading)
### What happened?

[The artist](https://twitter.com/SKULLZ_XXX) tried to mint his NFT on rmrk2. After minting it did not load. and he cannot find it in his profile: The webpage shows that he doesn't have that NFT. His account has been debited (indicating that the transaction was made), but there is nothing.

His NFT:

His Telegram: @skullz_xxx
Link to the NFT: https://kodadot.xyz/ksm/gallery/19449356-9692FA834AFCCE2D3E-BNG-GRL_1-00000001

### Please reproduce in steps

https://github.com/kodadot/nft-gallery/assets/104380776/fb5dc396-7854-4a33-9e07-cf0e7a10db14

### Expected Behavior

show minted NFt

### What browsers are you seeing the problem on?

_No response_

### At which address did you encounter bug?

_No response_

### Are you logged in?

None

### Which wallet you are using?

_No response_

### Screenshots

added

### Relevant log output

_No response_

### Payment link for reward

https://kodadot.xyz/ksm/transfer/?target=%3CMy_Kusama_Address_check_https://github.com/kodadot/nft-gallery/blob/main/CONTRIBUTING.md#creating-your-ksm-address%3E

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by daiagi**
**Thank you for your contribution** to the [KodaDot - One Stop Shop for Polkadot NFTs](https://kodadot.xyz).

👇 __ Let's make a quick check before the contribution.

## PR Type

- [x] Bugfix
- [ ] Feature
- [ ] Refactoring

## Context

- [x] Closes #7044
- [x] Closes #6916 

#### Did your issue had any of the "$" label on it?

- [x] My DOT address: [Payout](https://canary.kodadot.xyz/dot/transfer/?target=1cAsKZYNRb8dkSCpn4eVkEn6ycNZTGoDo5dGDgB8J1UUWK8)

### Testing (RMRK2 / KSM)
[My Account](https://canary.kodadot.xyz/ksm/u/DBVPJeM91LawsF8dqphFYmdPwtxfpXqbgBtVaxn41CT3Amn)
### mass mint 2 nfts + 1 listed for sell

[massmint - mint tx](https://kusama.subscan.io/extrinsic/19485763-5)
[massmint - list tx](https://kusama.subscan.io/extrinsic/19485777-2)

### Classic Mint - 2 copies + roaylty + list for  sell
[mint tx](https://kusama.subscan.io/extrinsic/19485812-4)
[list tx](https://kusama.subscan.io/extrinsic/19485820-3)

### Classic Mint - 1 copy + attributes + support
[mint + transfer batch tx](https://kusama.subscan.io/extrinsic/19485847-5)

:heavy_check_mark:  All Success

## Testing RMRK1
[collection](https://kodadot.xyz/rmrk/collection/1AD2CE70E162F5CC33-RUN)

### classic mint (2 copies  + list + attributes)
[mint tx](https://kusama.subscan.io/extrinsic/19486135-2)
[list tx](https://kusama.subscan.io/extrinsic/19486146-2)
[nft page](https://kodadot.xyz/rmrk/gallery/19486135-1AD2CE70E162F5CC33-RUN-POKEMON_1-0000000000000001)

### Massmint + list
[mint tx](https://kusama.subscan.io/extrinsic/19486187-2)
[list tx](https://kusama.subscan.io/extrinsic/19486192-2)

### Classic mint (1 copy +support tx)
[mint + support tx](https://kusama.subscan.io/extrinsic/19486222-2)

:heavy_check_mark:  All Success

### <samp>🤖 Generated by Copilot at 29e3900</samp>

Refactored and fixed the code for minting RMRK tokens in `transactionMintRmrk.ts`.

### <samp>🤖 Generated by Copilot at 29e3900</samp>

> _Sing, O Muse, of the skillful coder who refined_
> _The art of minting tokens with `system remarks` and `utility batch`,_
> _And who averted a grievous error that might have maligned_
> _The joyful message of success that the users should catch._

### CODE REVIEW

File: composables/transaction/mintToken/transactionMintRmrk.ts [hunk 1]
```diff
@@ -153,18 +149,15 @@ export async function execMintRmrk({
...
-    ? args
-    : [args.filter(Boolean).map((arg) => asSystemRemark(api, arg as string))]
+  const arg = [args]
 
   executeTransaction({
     cb,
     arg,
     successMessage:
-      item.successMessage ||
+      item.successMessage ??
```
Comment (roiLeo) [inline composables/transaction/mintToken/transactionMintRmrk.ts:160]: we have `item.successMessage ||` on other transaction file, should we change it there too?

**Review by roiLeo — COMMENTED** [review]
this part is quite sensitive,, maybe @prury can test it?

File: composables/transaction/mintToken/transactionMintRmrk.ts (hunk 1 above)
Comment (daiagi) [reply composables/transaction/mintToken/transactionMintRmrk.ts:160]: I was just following sonar lint "requests"
if you want ill open another PR to deal with that, so we will avoid mixing topics together

**Comment by prury** [comment]
checking

**Comment by prury** [comment]
| **RMRK1** |
| ------------- | 
| **Classic** : 3 copies, list for 1.2, 2 attributes ✔️ 
https://kusama.subscan.io/tx/0x5c9e8d9f84b387884cd77decd7d631c9d96d9efbce7de1481dd2defb0da23a0e
https://kusama.subscan.io/tx/0xa7b8f1cc2425d879341216a28c51d8a058f3d929632719a1e60e78e42ef9c552
https://deploy-preview-7082--koda-canary.netlify.app/rmrk/collection/7EA1DCF47E98A25067-HNO | 
| redirection ✔️ | 
|   -------- | 
| **Simple** : 2 copies, list for 2 ✔️ |
|  https://kusama.subscan.io/tx/0xbcf8d716f5f816cf9c66134a494d3e6e1c33b1a57749dc55f6b25c76ad32308e 
https://kusama.subscan.io/tx/0x8a4d255f6ba9b9c2d15bb58e3dff0aa6d5af901267e2c99d7b82ae3356423a87
https://deploy-preview-7082--koda-canary.netlify.app/rmrk/gallery/19489246-7EA1DCF47E98A25067-OCTO-OCTOPUS_DEVICE_1-0000000000000001
| redirection ✔️ | 
|   -------- | 
| **Mass** :  3 nfts, prices 1-3-5 ❌ |
https://kusama.subscan.io/tx/0x53d0796e188392866e51ad2749068ea299004acb319b906b27d8129f68ba2460
https://kusama.subscan.io/tx/0x593e2d70e007ed501d63de111b7bbcec796c2a3f552cd14f89e7ae181d1e26ee
https://deploy-preview-7082--koda-canary.netlify.app/rmrk/collection/7EA1DCF47E98A25067-MUSH
| redirection ✔️ | 

| **RMRK2**  | 
| ------------- | 
| **Classic** : Collection Creation, unlimited items
https://kusama.subscan.io/tx/0x70523ea48d6b0eb8038da81114de22b60e2bc123d329ace18326dfbf61e1a842 | 
|   -------- | 
| **Classic** : 2 copies, list for 2, 1 attribute, custom royalties recipient, 1% royalty
https://kusama.subscan.io/tx/0xbdbb9d3904aec4dc3150161038ade3b7a22680409d66ecec75526fe817c262f3
https://kusama.subscan.io/tx/0x2fd4017e93651d73db6b601db79fc706ffb022598a53e379f89fbe8230f0c5c2
https://deploy-preview-7082--koda-canary.netlify.app/ksm/gallery/19489462-7EA1DCF47E98A25067-CYBER-AI_CYBERPUNK_1_1-00000001
| redirection ✔️ | 
| ------------- | 
| **Mass** : 3 nfts, list for 2
https://kusama.subscan.io/tx/0x1f4e30629d9831d65668c0b9dfb69f658d140293b89a4848adb148cb4b8fc8f8
https://kusama.subscan.io/tx/0x9d950495790590197d2f526c7620b2cd9ccdff0f0567cb73b6b422bd85453509
https://deploy-preview-7082--koda-canary.netlify.app/ksm/collection/7EA1DCF47E98A25067-CYBER
| redirection ✔️ | 

side note: RMRK2 was strangely already working for me on canary before, proceed with caution

**Review by roiLeo — APPROVED** [review]

**Comment by daiagi** [comment]
>side note: RMRK2 was strangely already working for me on canary before, proceed with caution

I can explain
Prior to this PR Massmint was working and minting a single copy in classic mint also working
classic mint + copies > 1 => not working

**Comment by prury** [comment]
> > side note: RMRK2 was strangely already working for me on canary before, proceed with caution
> 
> I can explain Prior to this PR Massmint was working and minting a single copy in classic mint also working classic mint + copies > 1 => not working

ohh, allright, ty!

**Comment by daiagi** [comment]
image


oh @prury 
you got me a present? How lovely

**Comment by prury** [comment]
when testing only rmrk1 massmint failed, one time bug?

**Comment by prury** [comment]
massmint on RMRK1 worked properly with a new created collection:
https://kusama.subscan.io/tx/0xe121aaef865dec8255bec45d4accf98ecf6871f7c5929f8cb076df7e2522debe
https://kusama.subscan.io/tx/0x3143825b47378b4c7c6a7772abbee6886cc34c60f0d48433b0d1d5d40433c51d

**Comment by yangwao** [comment]
pay 50 usd

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $50 @ 4.26 USD/DOT ~ 11.737 $DOT
🧗 [1cAsKZYNRb8dkSCpn4eVkEn6ycNZTGoDo5dGDgB8J1UUWK8](https://polkadot.subscan.io/account/1cAsKZYNRb8dkSCpn4eVkEn6ycNZTGoDo5dGDgB8J1UUWK8)
🔗 [0x427536957de5a21a6081b2e1f8310d321ea3c0f0b9156488150af9eb3b111ece](https://polkadot.subscan.io/extrinsic/0x427536957de5a21a6081b2e1f8310d321ea3c0f0b9156488150af9eb3b111ece)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**